org.eclipse.scanning.sequencer/src/org/eclipse/scanning/sequencer/nexus/SolsticeScanMonitor.java

```diff
@@ -227,23 +245,51 @@ public void scanFinished() throws ScanningException {
 		// Note: we don't use scanFinally as that is called after the nexus file is closed.
 		final Dataset scanFinishedDataset = DatasetFactory.createFromObject(IntegerDataset.class, 1, null);
 		try {
-			this.scanFinished.setSlice(null, scanFinishedDataset,
+			this.scanFinishedDataset.setSlice(null, scanFinishedDataset,
 					new int[] { 0 }, new int[] { 1 }, new int[] { 1 });
 		} catch (Exception e) {
 			throw new ScanningException("Could not write scanFinished to NeXus file", e);
 		}
 		
-		Duration scanDurationObj = Duration.between(scanStartTime, Instant.now());
-		String scanDurationStr = durationInMillisToString(scanDurationObj);
+		Duration scanDuration = Duration.between(scanStartTime, Instant.now());
+		String scanDurationStr = durationInMillisToString(scanDuration);
 		logger.info("Scan finished in " + scanDurationStr);
 		
 		final Dataset scanDurationDataset = DatasetFactory.createFromObject(scanDurationStr);
 		try {
-			this.scanDuration.setSlice(null, scanDurationDataset,
+			this.scanDurationDataset.setSlice(null, scanDurationDataset,
 					new int[] { 0 }, new int[] { 1 }, new int[] { 1 });
 		} catch (Exception e) {
 			throw new ScanningException("Could not write scan duration to NeXus file", e);
 		}
+		
+		final long estimatedScanTimeMillis = model.getScanInformation().getEstimatedScanTime();
+		Duration scanDeadTime = scanDuration.minus(estimatedScanTimeMillis, ChronoUnit.MILLIS);
+		String scanDeadTimeStr = durationInMillisToString(scanDeadTime);
+		final Dataset scanDeadTimeDataset = DatasetFactory.createFromObject(scanDeadTimeStr);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "scanDeadTimeDataset" which hides the field declared at line 98. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AHiddenFieldCheck)